flint-spark-integration/src/main/scala/org/opensearch/flint/spark/FlintSparkIndex.scala

```diff
@@ -79,6 +79,30 @@ object FlintSparkIndex {
    */
   val ID_COLUMN: String = "__id__"
 
+  /**
+   * invalid index name characters.
+   */
+  val INVALID_INDEX_NAME_CHARS: Set[Char] =
+    Set(' ', ',', ':', '"', '*', '+', '/', '\\', '|', '?', '#', '>', '<')
+
+  /**
+   * Percent-encode invalid OpenSearch index name characters.
+   *
+   * @param indexName
+   *   raw Flint index name
+   * @return
+   *   percent-encoded index name
+   */
+  def percentEncode(indexName: String): String = {
+    indexName
+      .flatMap(ch =>
+        if (INVALID_INDEX_NAME_CHARS.contains(ch)) {
+          s"%${ch.toInt.toHexString}"
+        } else {
+          ch.toString
+        })
+  }
```

Comment:
Since string concatenation in a loop can be costly due to the immutability of strings in Scala, I would prefer using a mutable StringBuilder.

```suggestion
def percentEncode(indexName: String): String = {
  val encoded = new StringBuilder(indexName.length)
  for (ch <- indexName) {
    if (INVALID_INDEX_NAME_CHARS.contains(ch)) {
      encoded.append("%" + ch.toInt.toHexString.toUpperCase)
    } else {
      encoded.append(ch)
    }
  }
  encoded.toString()
}
```

Reply:
ack! will make the change. Thanks